src/main/java/de/bonndan/nivio/assessment/kpi/ScalingKPI.java

```diff
@@ -1,26 +1,92 @@
+ * yellow if scaled to 0 without relations
+ * orange of scaled to 0 as data sink
+ * unknown if no label or not a number
+ * green if scaled higher than 1
+ * yellow if a bottleneck (more than 1 item depend on it)
  */
-public class ScalingKPI extends CustomKPI {
+public class ScalingKPI extends AbstractKPI {
 
     public static final String IDENTIFIER = "scaling";
 
-    public ScalingKPI() {
-        label = Label.scale.name();
-        messageLabel = Label.scale.name();
-        ranges = Map.of(
-                Status.GREEN, Range.between(2d, 10d),
-                Status.YELLOW, Range.between(0d, 1d),
-                Status.RED, Range.between(0d, 0d)
-        );
+    private static final Logger LOGGER = LoggerFactory.getLogger(ScalingKPI.class);
+    public static final String SCALED_TO_ZERO = "scaled to zero";
+
+    @NonNull
+    @Override
+    public List<StatusValue> getStatusValues(Component component) {
+
+        if (!(component instanceof Item)) {
+            return List.of();
+        }
+
+        return getItemStatusValues((Item) component);
+    }
+
+    private List<StatusValue> getItemStatusValues(Item component) {
+
+        long usedAsDataTarget = component.getRelations(RelationType.DATAFLOW).stream()
+                .filter(relation -> relation.getTarget().equals(component))
+                .count();
+
+        long usedAsProvider = component.getRelations(RelationType.PROVIDER).stream()
+                .filter(relation -> relation.getSource().equals(component))
+                .count();
+
+        String scaleLabel = component.getLabel(Label.scale);
+        int scaleValue = -1;
+        if (scaleLabel != null) {
+            try {
+                scaleValue = Integer.parseInt(scaleLabel);
+                if (scaleValue == 0) {
+                    Status status = Status.YELLOW;
+                    String message = SCALED_TO_ZERO;
+                    if (usedAsProvider > 0) {
+                        status =  Status.RED;
+                        message += " and provider for " + usedAsProvider + " items";
+                    } else if (usedAsDataTarget > 0) {
+                        status = Status.ORANGE;
+                        message += " and data sink for " + usedAsDataTarget + " items";
+                    }
+                    return List.of(new StatusValue(Label.scale.name(), status, message));
+                }
+            } catch (NumberFormatException ignored) {
+                LOGGER.warn("Scaling KPI cannot handle label scale value '{}' of component '{}'", scaleLabel, component);
+            }
+        }
+
+        if (scaleValue == 1 && (usedAsProvider > 1)) {
+            return List.of(new StatusValue(Label.scale.name(), Status.YELLOW, String.format("Unscaled, but %d items depend on it.", usedAsProvider)));
+        }
+
+        if (scaleValue > 0) {
+            return List.of(new StatusValue(Label.scale.name(), Status.GREEN));
+        }
+
+        return List.of();
```

Comment:
I prefer `Collections.emptyList()`, consider a possible NPE as described here: https://stackoverflow.com/questions/39400238/list-of-or-collections-emptylist

Reply:
Wow, good to know. Fixed.